api/src/main/java/org/apache/iceberg/expressions/StrictMetricsEvaluator.java

```diff
@@ -402,6 +402,11 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       return ROWS_MIGHT_NOT_MATCH;
     }
 
+    @Override
+    public <T> Boolean notStartsWith(BoundReference<T> ref, Literal<T> lit) {
+      return ROWS_MIGHT_NOT_MATCH;
```

Comment:
This is correct, but I think there are cases where this can definitively conclude that rows cannot match. For example, `notStartsWith("x")` when the bounds are `["a", "b"]`.

Reply:
Cool. I had a note in my working branch about possible cases that could be handled, but I figured this PR is already large and so I figured that `ROWS_MIGHT_NOT_MATCH` will definitely give correct results and matches the surrounding code.

I'lll leave this comment up and open a follow up issue after addressing some of the comments here.

Reply:
I added a TODO in the code, similar to some of the other TODO's I've seen. Once this is merged I'll open an issue for a follow-up. Figure it's best not to open the issue until the issue exists in the master branch. 🙂

Reply:
There's a TODO - `// TODO: Handle cases that definitely cannot match, such as notStartsWith("x") when the bounds are ["a", "b"].`  - and once this is merged I'll be sure to open an issue.